### ISSUE TITLE
Ordering of module use statements

### ISSUE THREAD

**Issue by jrper** [FluidityProject/fluidity#87]
As I believe is common knowledge, the Intel Fortran compilers still appear to be very sensitive to the order of use statements in modules, with compiler errors liable to appear if these get too far away from the INTEL preferred order (if both a module and another which uses it are to be included, the parent must be included before the children).

Given that I'm currently running into problems with this while trying to use a branch on the imperial cluster, I'm proposing to fix this for as many modules as I need to, while also including a python script to detect this behaviour if this starts to be a problem for others. Since we also have external users running on intel, and others may have views on what they'd like to have happen I'm raising this as an issue now first.

**Comment by dham** [comment]
I don't have anything to offer other than sympathy. This has been an ifort bug for the best part of a decade now!

**Comment by jrper** [comment]
For anyone keep ing track at home branch 0868fdd27e899ef6ca1fbecc37427c69949b4a2b now has the work in it. Since my script accidentally ended up more powerful than I intended, I've applied it to everything. I'm now working on a test to go with this, then I'll open a pull request. If anyone has any idea how to deal with timing compiles under intel, I'd be interested whether this version is also faster, or  only avoids internal compiler errors.
